x/ibc/perm/types/extended.go

```diff
@@ -0,0 +1,53 @@
+package types
+
+import (
+	"slices"
+
+	"cosmossdk.io/errors"
+
+	"cosmossdk.io/core/address"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+func (rl *PermissionedRelayersList) HasRelayer(addr string) bool {
+	return slices.Contains(rl.Relayers, addr)
+}
+func (rl *PermissionedRelayersList) AddRelayer(addr string) {
+	rl.Relayers = append(rl.Relayers, addr)
+}
+
+func (rl *PermissionedRelayersList) GetAccAddr(cdc address.Codec) ([]sdk.AccAddress, error) {
+
+	var relayerStrs []sdk.AccAddress
+	for _, relayer := range rl.Relayers {
+
+		relayer, err := cdc.StringToBytes(relayer)
+		if err != nil {
+			return nil, errors.Wrapf(sdkerrors.ErrInvalidAddress, "string could not be parsed as address: %v", err)
+		}
+		relayerStrs = append(relayerStrs, relayer)
+	}
+
+	return relayerStrs, nil
+}
+
+func ToRelayerList(relayers []sdk.AccAddress) PermissionedRelayersList {
+	var relayerList PermissionedRelayersList
+	for _, relayer := range relayers {
+		relayerList.AddRelayer(relayer.String())
```

Comment:
We should avoid to use (sdk.AccAddress).String() in non-test code.

Reply:
have to use cdc.BytesToString()

Reply:
okay it's my fault